### DISCUSSION TITLE
Wind Modeling Runtime

### DISCUSSION THREAD

**Discussion by sebastianmezaurotadze** [firemodels/fds#13329] [Q&A] [answered]
Whenever I invoke wind through the "&WIND" line, it seems to make my simulations run slower compared to no wind, even if wind speed is set to 0. I think it has to do with how open boundary conditions aren't actually always modeled as open (See screenshot from the user guide)

Any ideas on how to make the open boundary conditions actually behave as open boundary conditions for time steps that the wind is set to 0?

![image](https://github.com/user-attachments/assets/c850b872-8acb-4618-9545-96878806b366)

**Comment by rmcdermo** [comment]
Please post a simple example of the timing difference you are seeing.  I don't see why there would be a significant time difference.  Probably, there is an implicit DT restriction due to a velocity profile or something that is causing the longer run times.

**Reply by sebastianmezaurotadze** [reply]
Here is an example comparison of wind in an empty domain. The simulation starts with no wind until it ramps to 20 m/s from 50 -> 51 s

As you can see, the version with wind specified on the "&WIND" line runs significantly more slowly when wind speed is set to 0 m/s before the wind gust, but once the wind is enabled both run at the same speed.

![total_time_vs_seconds_passed](https://github.com/user-attachments/assets/fe9f04b6-e550-48a6-9a0d-9b803d7375f0)

I've also included  some additional plot and the FDS code below:


![max_pressure_error_vs_seconds_passed](https://github.com/user-attachments/assets/9f984b28-646d-48e5-8f4c-4c33256de973)
![max_velocity_error_vs_seconds_passed](https://github.com/user-attachments/assets/4e3ed9b1-e718-49b8-bfb0-d7b10b558bd4)
![pressure_iterations_vs_seconds_passed](https://github.com/user-attachments/assets/35afec00-8b75-4df5-b558-f6b0e481e0bc)
![step_size_vs_seconds_passed](https://github.com/user-attachments/assets/21683459-7f39-497a-9714-15940ccb61bb)
![time_step_vs_seconds_passed](https://github.com/user-attachments/assets/045b8757-3b79-4ee5-be75-8a865a32e34b)


FDS Code:
[Global_Wind_FDS.txt](https://github.com/user-attachments/files/16663730/Global_Wind_FDS.txt)
[Wall_of_Wind_FDS.txt](https://github.com/user-attachments/files/16663731/Wall_of_Wind_FDS.txt)

**Reply by sebastianmezaurotadze** [reply]
Is there any way to relax the DT restriction to improve runtime? I would prefer to use global wind as opposed to wall of wind since my end goal is to model wind gusts into a domain from different direction at different time stamps.

**Reply by mcgratta** [reply]
I'll look at the case.

**Comment by mcgratta** [comment]
Setting the wind to 0 m/s might cause the boundaries to appear as solid walls. I don't know why that would make the simulation slower. In any case, look at the .out file and look at what the typical time step size is. Then look to see where in the domain the highest flow sped is achieved. That usually is a clue to why a job is running slow.

**Comment by mcgratta** [comment, marked as answer]
FDS uses the nominal wind speed in its calculation of the initial time step. In your case, during the ramp up stage, the CFL number is relatively low. Try setting your own initial time step, `DT`, on the `TIME` line.

**Reply by sebastianmezaurotadze** [reply]
Resolved my issue, thank you!
